nodes/Nats.service.ts

```diff
@@ -34,11 +51,23 @@ export class NatsService {
 		let entry = this.connections.get(cid)
 
 		if(entry && entry.connection.isClosed()) {
-			entry = undefined //reconnect
+			// Remove the closed connection from cache and create a new one
+			this.connections.delete(cid)
+			entry = undefined
 		}
 
 		if (!entry) {
 			const connection = await connect(options)
+			
+			// Add connection event listeners for better reconnection handling
+			connection.closed().then(() => {
```

Comment:
I had issues with the internal nats client connection state management in the past. Maybe this solves the issue.